sdk/tables/azure-data-tables/azure/data/tables/_error.py

```diff
@@ -61,9 +61,9 @@ def _wrap_exception(ex, desired_type):
 
 
 def _validate_table_name(table_name):
-    if match("^[a-zA-Z]{1}[a-zA-Z0-9]{2,62}$", table_name) is None:
+    if table_name is None:
```

Comment:
This is doing the same check for table_name as in [_table_client.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/azure/data/tables/_table_client.py#L78) and [_table_client_async.py](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/tables/azure-data-tables/azure/data/tables/aio/_table_client_async.py#L76), you can remove this function.